server/app/models/grid_service.rb

```diff
@@ -147,7 +147,7 @@ def stack_exposed?
   end
 
   def deploy_pending?
-    self.grid_service_deploys.where(started_at: nil).count > 0
+    self.grid_service_deploys.where(finished_at: nil).count > 0
```

Comment:
This may run into problems with stale `GridServiceDeploy`s where a `GridServiceSchedulerWorker` has set `started_at` (causing other workers to ignore it), but the worker fails to either perform the deploy (such as #2213) or un-set `started_at` (like if the server crashes during the blocked-deploy operation).

AFAIK this means that there could be any number of really old stale `GridServiceDeploy`s that match this query, because AFAIK we don't clean out old/stale `GridServiceDeploy`s?